pkg/apis/sources/v1alpha1/apiserver_types.go

```diff
@@ -87,7 +87,7 @@ type ApiServerSourceSpec struct {
 
 	// Sink is a reference to an object that will resolve to a domain name to use as the sink.
 	// +optional
-	Sink *corev1.ObjectReference `json:"sink,omitempty"`
+	Sink *apisv1alpha1.Destination `json:"sink,omitempty"`
```

Comment:
FWIW, the duck doesn't use a pointer: https://github.com/knative/pkg/blob/a9458a30c0bbbed59420c5ea0e08187e697f4267/apis/duck/v1/source_types.go#L54

In fact, should we just embed the duck into the spec, rather than adding Sink (and CloudEventOverrides) directly?

Reply:
it does not, and `// +optional` is incorrect. You need a sink.

Reply:
I'm not sure why this was ever optional, but I don't want to change it so close to the release. Let's do a more thorough investigation after the cut.

Reply:
I remember why... in the OG container source, I allowed for sink to be passed as a command line flag OR a envvar.